components/binary_sensor/xiaomi.py

```diff
@@ -108,7 +121,17 @@ def is_on(self):
         """Return true if sensor is on."""
         return self._state
 
+    @property
+    def device_state_attributes(self):
+        attrs = {ATTR_OPEN_SINCE: self._open_since}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
     def parse_data(self, data):
+        if NO_CLOSE in data:  # handle push from the hub
+            self._open_since = data[NO_CLOSE]

```

Comment:
Replace it with self._device_state_attributes[ATTR_OPEN_SINCE] = data[NO_MOTION].
_device_state_attributes XiaomiDevice

Then you can remove device_state_attributes function in this class.